File: frontend/grazetrack_app/lib/features/expenses/models/expense_model.dart
class ExpenseModel {
  final String id;
  final String userId;
  final String type;
  final String description;
  final double amount;
  final String animalId;
  final String date;

  const ExpenseModel({
    required this.id,
    required this.userId,
    required this.type,
    this.description = '',
    required this.amount,
    this.animalId = '',
    required this.date,
  });

  factory ExpenseModel.fromJson(Map<String, dynamic> json) => ExpenseModel(
        id: json['id'] ?? '',
        userId: json['userId'] ?? '',
        type: json['type'] ?? '',
        description: json['description'] ?? '',
        amount: (json['amount'] ?? 0).toDouble(),
        animalId: json['animalId'] ?? '',
        date: json['date'] ?? '',
      );

  Map<String, dynamic> toJson() => {
        'type': type,
        'description': description,
        'amount': amount,
        'animalId': animalId,
        'date': date,
      };
}

File: backend/src/models/expenseModel.js
/**
 * Expense record data structure for Firestore
 * Collection: expenses/{expenseId}
 */
const ExpenseModel = {
  userId: "",       // Who recorded this expense
  type: "",         // feed | medicine | labor | equipment | other
  description: "",  // What the expense was for
  amount: 0,        // Amount spent
  animalId: "",     // Optional: link to a specific animal
  receipt: "",      // Optional: Firebase Storage URL for receipt photo
  date: "",         // ISO timestamp of when expense occurred
  createdAt: "",
};

module.exports = ExpenseModel;
